### PR TITLE
Lambdas: deep indexing for PDF

### PR DESCRIPTION
**Pull request by akarve**
- [x] Unit tests
- [x] Documentation
- [x] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

**Review by kevinemoore — APPROVED** [review]
A few docs comments, but looks good.

File: lambdas/es/indexer/index.py [hunk 1]
```diff
@@ -486,6 +503,22 @@ def maybe_get_contents(bucket, key, ext, *, etag, version_id, s3_client, size):
     return content
 
 
+def extract_pdf(file_):
```
[the remark this answers is not in the document]
Comment (akarve) [reply lambdas/es/indexer/index.py:506]: Yes there is. We just kind of YOLO that risk. I could head the file and check the size ahead of time.

[the remark this answers is not in the document]
Comment (akarve) [reply lambdas/es/indexer/index.py:506]: In all likelihood, up to a factor of 2, we've already run out of memory before this.

[the remark this answers is not in the document]
Comment (kevinemoore) [reply lambdas/es/indexer/index.py:506]: Do we limit the download size to our memory limit? Or, just try to grab the whole file no matter what?

[the remark this answers is not in the document]
Comment (akarve) [reply lambdas/es/indexer/index.py:506]: In most cases, incl. ones mentioned above, we grab the whole thing. For Parquet and Jupyter there's no progressive parsing lib so, until and unless we use s3fs, we are stuck grabbing the whole thing.

**Pull request merged**